### DISCUSSION TITLE
Can't change the 'source' entity of threshold sensor

### DISCUSSION THREAD

**Discussion by WayneManion** [home-assistant/frontend#16721] [Configuration]
This is a pretty minor issue, but here goes. It is definitely a first-world problem. 

Let's say I have created a threshold sensor. In this case, I was using a Tasmota power meter to know whether or not a dumb device (a countertop icemaker) is actually doing work. When I create the helper entity using Lovelace I can specify the source of the data used for the threshold sensor. 

As I refined the operation of this threshold sensor I could change the hysteresis, upper limit, and the lower limit within Lovelace. That is great. The issue for me is that when I upgraded to a Zigbee power meter I could not change the source of the input data for the threshold sensor. I had to delete the sensor and start over. When I deleted the sensor, some of my automations quit working, even though I re-added the threshold sensor seconds later. 

I would like to be able to change the source of input data for a threshold sensor in the same way I can change the hysteresis and the upper and lower limits. I haven't done an exhaustive check, but I think some other helper entity types may have this same issue. 

While we're going nuts asking for things, it would be awesome to be able to add new template entities from the helper entity UI.

**Comment by goedh452** [comment]
And maybe display the input sensor in the attributes field so it's easy to see which sensor is used.

**Comment by ic-dev21** [comment]
This would be nice to be able to do.

**Comment by karwosts** [comment]
This is a core issue, not a frontend issue. The integration itself defines which settings are changable after creation using an OptionsFlow. 

It's not up to frontend to choose which options are given.

**Comment by L0bit0** [comment]
do we have a solution for this? even if is not on front end, do we know yaml file where this is defined?

**Comment by amosstorkey** [comment]
Regardless of whether it is changeable, the front end should _show_ what the dependent sensor is. It doesn't even show that, so for a threshold, there is no way of visually knowing what the threshold is defined on (without digging into the helpers yaml).

**Comment by fxstein** [comment]
The lack of being able to change the source sensor really limits the use of this type of helper. Especially when used on top of sensors from various integrations. A single change at the source means redoing the helper and everything that goes with it. (like tags)

**Comment by karwosts** [comment]
Coming soon: https://github.com/home-assistant/core/pull/119157
